src-tauri/src/node.rs

```diff
@@ -57,35 +61,51 @@ impl NativeExecutionDispatch for ExecutorDispatch {
 /// if there is a node instance running previously,
 /// first it stops the previous instance, then starts a new instance
 pub(crate) async fn start_node(path: String, node_name: String) -> bool {
+    type FullClient<RuntimeApi, ExecutorDispatch> =
+        sc_service::TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<ExecutorDispatch>>;
+
     static NODE_HANDLE: Mutex<Option<JoinHandle<()>>> = Mutex::const_new(None);
+    static WEAK_CLIENT: Mutex<Option<Weak<FullClient<RuntimeApi, ExecutorDispatch>>>> =
+        Mutex::const_new(None);
 
     let mut node_handle_guard = NODE_HANDLE.lock().await;
+    let mut weak_client_guard = WEAK_CLIENT.lock().await;
 
     // if there is already a node running, stop it
-    if let Some(guard) = node_handle_guard.take() {
-        guard.abort();
+    if let Some(handle) = node_handle_guard.take() {
+        handle.abort();
+        while !handle.is_finished() {
+            sleep(Duration::from_millis(200)).await;
+        }
+        let weak_client = weak_client_guard.take().unwrap(); // safe to unwrap, since we know a previous instance is present
```

Comment:
If safe then use `.expect()`?

Reply:
Isn’t the only difference between `expect` and `unwrap` is, 
`expect` = `unwrap` + message ?

Why should I prefer `expect` over `unwrap` if safe?
If it was not safe, i shouldn't have `unwrapped` at all, and handled the error instead.

Reply:
`.expect()` forces you to provide an explicit message that convinces reviewer why it is okay to do that. We don't use `.unwrap()` at all in the node's code and we plan to eliminate it in farmer's code too. Substrate has the same policy.

Reply:
now I understood :)